src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/ProjectMigrationReport.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Xproj
+{
+    /// <summary>
+    /// Migration report holder. Copied from
+    /// https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.ProjectJsonMigration/Report/ProjectMigrationReport.cs
+    /// </summary>
+    internal class ProjectMigrationReport
+    {
+        [JsonProperty]
+        public string ProjectDirectory { get; private set; }
+
+        [JsonProperty]
+        public string ProjectName { get; private set; }
+
+        [JsonProperty]
+        public string OutputMSBuildProject { get; private set; }
+
+        [JsonProperty]
+        public List<MigrationError> Errors { get; private set; }
+
+        [JsonProperty]
+        public List<string> Warnings { get; private set; }
```

Comment:
Still mutable. I believe JSON.NET supports collection interfaces, so you could use `IReadOnlyList<string>`